src/python/pants/option/parser_hierarchy.py

```diff
@@ -15,6 +15,23 @@ def enclosing_scope(scope):
   return scope.rpartition('.')[0]
 
 
+def all_enclosing_scopes(scope, allow_global=True):
```

Comment:
Adding a few tiny tests for this method would be good.

Reply:
Agreed, I've added them in a [new file](https://github.com/cosmicexplorer/pants/blob/195d2ca90138417e119b1239557f05cb671263d4/tests/python/pants_test/option/test_enclosing_scopes.py) in 195d2ca. The tests I added are more than a few lines, but I can't immediately see how to reduce them further without dropping meaningful assertions.